packages/esbuild/src/executors/esbuild/lib/build-esbuild-options.ts

```diff
@@ -123,6 +127,50 @@ export function buildEsbuildOptions(
   return esbuildOptions;
 }
 
+/**
+ * When using TS project references we need to map the paths to the referenced projects.
+ * This is necessary because esbuild does not support project references out of the box.
+ * @param context ExecutorContext
+ */
+export function createPathsFromTsConfigReferences(
+  context: ExecutorContext
+): Record<string, string[]> {
+  const {
+    findAllProjectNodeDependencies,
+  } = require('nx/src/utils/project-graph-utils');
+
+  const deps = findAllProjectNodeDependencies(context.projectName);
+  const tsConfig = readJsonFile(
+    joinPathFragments(context.root, 'tsconfig.json')
+  );
+  const referencesAsPaths = new Set(
+    tsConfig.references.map((ref) => joinPathFragments(workspaceRoot, ref.path))
+  );
+
+  // for each dep we check if it contains a build target
+  // we only want to add the paths for projects that do not have a build target
+  return deps.reduce((acc, dep) => {
+    const projectNode = context.projectGraph.nodes[dep];
+    if (!projectNode.data.targets?.build) {
```

Comment:
Can we do anything about the hardcoded `build` task name? Folks might have different build task names.